spec/filters/bugs/file.rb

```diff
@@ -0,0 +1,12 @@
+opal_filter "File" do
+  fails "File.join returns a duplicate string when given a single argument"
+  #fails "File.join respects the given separator if only one part has a boundary separator"
+  #fails "File.join prefers the separator of the right part if both parts have separators"
+  fails "File.join raises a TypeError exception when args are nil"
+  fails "File.join calls #to_str"
+  fails "File.join calls #to_path"
+  fails "File.join flattens nested arrays"
+  fails "File.join raises an ArgumentError if passed a recursive array"
+  fails "File.join inserts the separator in between empty strings and arrays"
+  fails "File.join doesn't mutate the object when calling #to_str"
```

Comment:
If this spec relies on mutable strings then it should be in `unsupported/file.rb`